### DISCUSSION TITLE
Add support for reCAPTCHA v2 Invisible & v3

### DISCUSSION THREAD

**Discussion by kjmartens** [solspace/craft-express-forms#58] [Feature Requests]
Add support for reCAPTCHA v2 Invisible & reCAPTCHA v3, in addition to reCAPTCHA v2 Checkbox.

**Comment by terryupton** [comment]
Any plans for this still?
I’m seeing lots of spam submissions even with recaptcha enabled and setup. Any ideas on how to resolve? I can send some examples if required.

**Comment by kjmartens** [comment]
We hope to add this sometime in the future still but could be a few months out.

**Comment by terryupton** [comment]
Ok thanks. Any ideas on why the current google captcha isn’t being effective?

**Comment by kjmartens** [comment]
No idea... I've never heard of reCAPTCHA not being effective, and I haven't heard of anyone report issues with it either...

**Comment by kevadamson** [comment]
+1

**Comment by RuthAdele** [comment]
Any reason why this hasn't been added yet?
It's a pretty simple configuration option when setting up reCaptcha - I can't see why it shouldn't use invisible by default..?

**Comment by alexr8** [comment]
Hi, are there any plans to implement v3? Seeing quite a lot of spam submissions across a few sites running express forms using recaptcha v2 + honeypot. Would be great to upgrade them to recaptcha v3.

**Comment by kjmartens** [comment]
Sorry, we don't have any timeframe on when this might be available.

**Comment by thijskaspers** [comment]
Same here, getting loads of spam daily with honeypot and ReCaptcha v2 enabled. ReCaptcha v2 isn't good anymore in preventing spam, and Google isn't fixing it as they tell you to upgrade to v3.

This makes Express Forms not a good solution for forms anymore unfortunately.

**Comment by alexr8** [comment]
I might be wrong but it appears to be something specific to Express forms implementation of the v2 recaptcha - I switched one site over to sprout forms (the free tier), which uses v2 recaptcha as well, and i haven't had any spam since.

**Comment by kjmartens** [comment]
@thijskaspers @alexr8 Sorry to hear about this. I've just played around with reCAPTCHA v2 Checkbox on an Express Forms form (acted "spammy" and spammed it for a while) and it eventually started challenging me. I can't imagine how reCAPTCHA v2 Checkbox could vary from one form plugin to another. Just to confirm, you did set it up correctly and see the special Checkbox show up in the form on the front end?

I will however, see if we can add support for v2 Invisible and v3 in the near future. 🙂

**Comment by mikesnoeren** [comment]
Any updates on this? We have a few sites that rely on express forms, but v2 is really not doing a good job.

**Reply by kjmartens** [reply]
@mikesnoeren We will be making an announcement about this within the next few weeks. 🙂

**Comment by mikesnoeren** [comment]
Unless there is a way to add Captcha 3 to Express Forms, we will be uninstalling Express Forms soon. 
Does anyone have any update on this?

**Comment by thijskaspers** [comment]
Waiting for an update too! ReCaptcha v2 has no effect anymore

**Comment by kjmartens** [comment]
Unfortunately, Express Forms has now been discontinued. You can read more about it here:
[Important Notice - Express Forms Discontinued](https://github.com/solspace/craft-express-forms/issues/126)

We apologize for any inconvenience caused.
